### PR TITLE
Emsdk

### PR DESCRIPTION
**Pull request by dnfield**
Pulls down emsdk as part of gclient sync _if_ download_emsdk is set to true.

I had started to add a WASM build target for this as well, but it's taking a lot to get that working so backing it out for now. Leaving in the gn args for instructional purposes, but could remove that as well if it's distracting.

@yjbanov @hterkelsen @sbc100

### CODE REVIEW

File: DEPS [hunk 1]
```diff
@@ -606,6 +613,29 @@ hooks = [
...
+  {
+    'name': 'emsdk activate',
+    'pattern': '.',
+    'condition': 'download_emsdk',
+    'action': [
+      'python',
+      'src/third_party/emsdk/emsdk.py',
+      'activate',
+      'latest',
+      '--embedded', # Makes the installation hermetic to //third_party/emsdk
```
Comment (sbc100) [inline DEPS:636]: You don't need `--embedded` anymore, its the default.

File: DEPS [hunk 2]
```diff
@@ -606,6 +613,29 @@ hooks = [
       'src/third_party/dart/third_party/7zip.tar.gz.sha1',
     ],
   },
+  {
+    'name': 'emsdk install',
+    'pattern': '.',
+    'condition': 'download_emsdk',
+    'action': [
+      'python',
+      'src/third_party/emsdk/emsdk.py',
```
Comment (sbc100) [inline DEPS:622]: Maybe you want to control the version of python you are using but `emsdk.sh` is the recommended entry point.

Comment (dnfield) [reply DEPS:622]: This needs to run on windows as well as Mac and Linux.  Plus gclient is already python

File: DEPS (hunk 1 above)
Comment (dnfield) [reply DEPS:636]: Do you know what revision?  We have to roll forward to at least that rev.

Comment (sbc100) [reply DEPS:636]: It was revision 858b176 from about 3 months ago.

**Comment by yjbanov** [comment]
Closing the PR so it doesn't pop up in the review queue. It's still a good reference for the future, when/if we decide to build CanvasKit on Flutter's infra.

**Pull request closed without merging**